lib/AboutPreferences.jsm

```diff
@@ -257,6 +260,36 @@ this.AboutPreferences = class AboutPreferences {
       });
     });
 
+    try {
+      const discoveryStreamPref = JSON.parse(ASPreferences.values["discoverystream.config"]);
+      if (discoveryStreamPref.enabled) {
```

Comment:
Seems like we should probably avoid parsing the pref here. @ScottDowne would it be safe to check `state.DiscoveryStream.layout` to be non-0 length? Or even DiscoveryStream.config.enabled would avoid re-parsing json. ??

Reply:
`DiscoveryStream.config.enabled` would probably be ideal. I dunno if checking the length might be weird.